enterprise/server/cmd/ci_runner/main.go

```diff
@@ -187,10 +210,61 @@ type workspace struct {
 	log io.Writer
 }
 
+func newArtifactUploader(ctx context.Context, bep *build_event_publisher.Publisher, cacheTarget, instanceName string) (*bes_artifact_uploader.Uploader, error) {
+	u, err := url.Parse(cacheTarget)
+	if err != nil {
+		return nil, status.InvalidArgumentErrorf("failed to parse cache target %q: %s", cacheTarget, err)
+	}
+	conn, err := grpc_client.DialTarget(cacheTarget)
+	if err != nil {
+		return nil, err
+	}
+	// Note: we don't defer conn.Close() here since it's OK to keep it alive
+	// until the CI runner exits.
+	bsClient := bspb.NewByteStreamClient(conn)
+	ul := bes_artifact_uploader.New(ctx, bsClient, bep, u.Host, instanceName)
```

Comment:
this code could all move to uploader.New

Reply:
Done - I was on the fence about having the uploader do a Dial operation in the constructor but I think it doesn't really matter for now. Done